lib/screen/detail/preview/artwork_preview_page.dart

```diff
@@ -529,11 +530,19 @@ class _ArtworkPreviewPageState extends State<ArtworkPreviewPage>
                     physics: isFullScreen
                         ? const NeverScrollableScrollPhysics()
                         : null,
-                    onPageChanged: (value) {
+                    onPageChanged: (value) async {
                       final currentId = widget.payload.identities[value];
                       _bloc.add(ArtworkPreviewGetAssetTokenEvent(currentId));
                       _stopAllChromecastDevices();
                       keyboardManagerKey.currentState?.hideKeyboard();
+                      final mixPanelClient = injector.get<MixPanelClientService>();
+                      await mixPanelClient.trackEvent(
```

Comment:
we can allow these calls `trackEvent` to do parallel in the background, and ignore if it errors, so I think you can remove the `await` here.

Please apply for all.

Reply:
Also, could you please help to change the `metricClient.addEvent` if it was impelemented the same?